client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/VectorLayers/VectorLayer.cs

```diff
@@ -138,9 +138,7 @@ await QueuedTask.Run(() =>
 
     public async Task<bool> InitializeEventsAsync()
     {
-      bool result = Layer.ConnectionStatus == ConnectionStatus.Connected;
-
-      if (result)
+      if (Layer.ConnectionStatus == ConnectionStatus.Connected)
```

Comment:
if (Layer.ConnectionStatus != ConnectionStatus.Connected)
{
    return false;
 }

Reply:
We can't return there yet because we have a call to LoadMeasurementsAsync(), which is being invoked in any case.